composer/metrics/metrics.py

```diff
@@ -53,8 +53,8 @@ def update(self, logits: Tensor, targets: Tensor):
             area_prediction = torch.histc(pred.float(), bins=self.num_classes, min=0, max=self.num_classes - 1)
             area_target = torch.histc(target.float(), bins=self.num_classes, min=0, max=self.num_classes - 1)
 
-            self.total_intersect += area_intersect
-            self.total_union += area_prediction + area_target - area_intersect
+            self.total_intersect += area_intersect  # type: ignore
```

Comment:
there are a lot of type ignores added. should i be worried about that?

Reply:
I can add some type assertions in here instead:
```
assert isinstance(self.total_intersect, Tensor)
assert isinstance(self.total_union, Tensor)
self.total_intersect += area_intersect
self.total_union += area_prediction + area_target - area_intersect
```
instead since the issue that's reported is 
```
Operator "+" not supported for types "Module" and "Tensor" (reportGeneralTypeIssues)
```

Reply:
yes, throughout, if you _know_ the type of the thing and just need to tell the type checker, assertion is preferred over type ignore, because its more clear for the reader and validates the invariant rather than ignoring it.

Reply:
Added this fix to most of the `# type: ignore` changes, but I left it for changes that seem a bit esoteric/confusing to read or invalid (e.g. the case of the remaining `test_fsdp.py` ignores and also the ignores related to module parameters equalling `None`)